users/views.py

```diff
@@ -39,10 +40,28 @@ def post(self, request):
                 name           = data['name'],
                 email          = data['email'],
                 password       = decoded_hashed_password,
-                is_sms_agree   = data['is_sms_agree'],
-                is_email_agree = data['is_email_agree'],
+                is_sms_agree   = data['smscheck'],
+                is_email_agree = data['emailcheck'],
             )
             return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
 
         except KeyError:
-            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
\ No newline at end of file
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+class LoginView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if not (data['email'] and data['password']):
+                return JsonResponse({"MESSAGE":"EMPTY_VALUE_ERROR"}, status=400)
+
+            if User.objects.filter(email=data['email']).exists():
+                if bcrypt.checkpw(data['password'].encode('utf-8'), User.objects.get(email=data['email']).password.encode('utf-8')):
+                    access_token = jwt.encode({'id':User.objects.get(email=data['email']).id}, SECRET_KEY, algorithm=ALGORITHM)
+                    return JsonResponse({"MESSAGE":"SUCCESS","ACCESS_TOKEN":access_token}, status=201)
+            
```

Comment:
`User.objects.get(email=data['email'])` 코드가 중복되네요. 중복되는 문제도 있지만, `objects.get()` method를 호출할 때마다 데이터베이스에 요청을 보냅니다. 그렇기 때문에 아래와 같이 한번만 호출해서 데이터베이스에서 가져온 값을 user 변수에 저장(메모리에 올려놓고)해서 사용할 수 있도록 수정해주세요~

```suggestion
            if not User.objects.filter(email=data['email']).exists():
                return JsonResponse({"MESSAGE":"INVALID_USER"}, status=401)
            user = User.objects.get(email=data['email'])
            
            if bcrypt.checkpw(..):
                    access_token = jwt.encode({'id':user ..}, SECRET_KEY, algorithm=ALGORITHM)
                    return JsonResponse({"MESSAGE":"SUCCESS","ACCESS_TOKEN":access_token}, status=201)
            
```

Reply:
알려주신 방식으로 user에 정보 저장하여 사용하는 방법으로 수정하였습니다!!
많은 가르침 감사드립니다 :)